text/src/autogluon/text/automm/predictor.py

```diff
@@ -356,11 +349,19 @@ def fit(
         else:  # continuing training
             model = self._model
 
-        val_metric, minmax_mode = get_metric(
-            metric_name=self._eval_metric_name,
-            num_classes=output_shape
+        if self._validation_metric_name is None:
+            validation_metric_name = infer_validation_metric(
+                problem_type=problem_type,
+                eval_metric_name=self._eval_metric_name,
+            )
+        else:
+            validation_metric_name = self._validation_metric_name
+
+        validation_metric, minmax_mode = get_metric(
+            metric_name=validation_metric_name,
+            num_classes=output_shape,
         )
-        custom_metric_func = get_custom_metric_func(self._eval_metric_name)
+        custom_metric_func = get_custom_metric_func(validation_metric_name)
```

Comment:
I feel that this function can be combined into `get_metric`. Basically, for each validation_metric_name, it will return the aggregator and the custom_metric_func.

Reply:
If merged, most metrics need to return None for custom_metric_func, which seems weird.

Reply:
In the current implementation, the `get_metric` and `get_custom_metric_func` will be highly coupled. If you are going to add a new metric, you will need to change the implementation of both functions. Instead, you may just implement the 'custom_metric_func' feature in `get_metric` so the future contributors will only need to revise the `get_metric` function.

Reply:
Merged.